goreleaser attempts to push to snap store when publish is set to false

**Describe the bug**

goreleaser attempts to push to snap store, even if `publish` is set to false in the snapcraft configuration

**To Reproduce**

Steps to reproduce the behavior:

```yaml
project_name: f1viewer
builds:
  - 
    goos:
      - linux
      - windows
      - darwin
    goarch:
      - amd64
      - 386
archives:
  -
    format: binary
    replacements:
      amd64: 64-bit
      386: 32-bit
      darwin: macOS
    files:
      - none*
checksum:
  name_template: "checksums.txt"
snapshot:
  name_template: "{{ .Tag }}-next"
changelog:
  sort: asc
  filters:
    exclude:
    - '^docs:'
    - '^test:'
    - Merge pull request
    - Merge branch
nfpms:
  -
    description: TUI client for F1TV
    license: GPL-3.0
    homepage: https://github.com/SoMuchForSubtlety/f1viewer/
    maintainer: SoMuchForSubtlety <s0muchfrsubtlety@gmail.com>
    replacements:
      darwin: macOS
    formats:
      - deb
      - rpm
    dependencies:
      - mpv
    recommends:
      - xclip
snapcrafts:
  -
    replacements:
      darwin: macOS
    publish: false
    summary: TUI client for F1TV
    description: |
      F1viewer is a terminal F1TV client that lets you watch F1 with mpv
      or any other player that supports HLS streams.
    grade: stable
    confinement: classic
    license: GPL-3.0
    base: core18
brews:
  -
    description: TUI client for F1TV
    homepage: https://github.com/SoMuchForSubtlety/f1viewer/
    folder: Formula
    github:
      owner: SoMuchForSubtlety
      name: homebrew-tap
    commit_author:
      name: goreleaserbot
      email: goreleaser@carlosbecker.com
    dependencies:
      - mpv
```

**Expected behavior**

Publishing to the snap store is skipped.

**Environment (please complete the following information):**

[executed in a GitHub action](https://github.com/SoMuchForSubtlety/f1viewer/blob/6f24dc0599344963d46d31ad69ff8175c7564d2d/.github/workflows/release.yml)

**Additional context**
```
[...]
       • SNAPCRAFT PACKAGES       
         • pushing snap              snap=f1viewer_1.0.0_linux_amd64.snap
         • pushing snap              snap=f1viewer_1.0.0_linux_386.snap
   ⨯ release failed after 77.27s error=Snapcraft Packages: failed to publish artifacts: failed to push dist/f1viewer_1.0.0_linux_386.snap package: Preparing to push 'f1viewer_1.0.0_linux_386.snap'.
After pushing, the resulting snap revision will be released to 'stable' when it passes the Snap Store review.
Install the review-tools from the Snap Store for enhanced checks before uploading this snap.
Pushing 'f1viewer_1.0.0_linux_386.snap' [================================] 100%
Received:
- macaroon-permission-required: Permission "channel" is required as a macaroon caveat.
##[error]The process '/home/runner/work/_temp/56153f7c-7f35-4ffe-8488-507790be17a4/goreleaser' failed with exit code 1
```

--- a/internal/pipe/snapcraft/snapcraft.go
+++ b/internal/pipe/snapcraft/snapcraft.go
@@ -141,6 +141,9 @@ func isValidArch(arch string) bool {
 
 // Publish packages
 func (Pipe) Publish(ctx *context.Context) error {
+	if ctx.SkipPublish {
+		return pipe.ErrSkipPublishEnabled
+	}
 	snaps := ctx.Artifacts.Filter(artifact.ByType(artifact.PublishableSnapcraft)).List()
 	var g = semerrgroup.New(ctx.Parallelism)
 	for _, snap := range snaps {

--- a/internal/pipe/snapcraft/snapcraft_test.go
+++ b/internal/pipe/snapcraft/snapcraft_test.go
@@ -7,10 +7,9 @@ import (
 	"path/filepath"
 	"testing"
 
-	"github.com/goreleaser/goreleaser/internal/testlib"
-
 	"github.com/goreleaser/goreleaser/internal/artifact"
 	"github.com/goreleaser/goreleaser/internal/pipe"
+	"github.com/goreleaser/goreleaser/internal/testlib"
 	"github.com/goreleaser/goreleaser/pkg/config"
 	"github.com/goreleaser/goreleaser/pkg/context"
 	"github.com/stretchr/testify/assert"

--- a/internal/pipe/snapcraft/snapcraft_test.go
+++ b/internal/pipe/snapcraft/snapcraft_test.go
@@ -7,6 +7,8 @@ import (
 	"path/filepath"
 	"testing"
 
+	"github.com/goreleaser/goreleaser/internal/testlib"
+
 	"github.com/goreleaser/goreleaser/internal/artifact"
 	"github.com/goreleaser/goreleaser/internal/pipe"
 	"github.com/goreleaser/goreleaser/pkg/config"
@@ -353,6 +355,19 @@ func TestPublish(t *testing.T) {
 	})
 	err := Pipe{}.Publish(ctx)
 	assert.Contains(t, err.Error(), "failed to push nope.snap package")
+}
+
+func TestPublishSkip(t *testing.T) {
+	var ctx = context.New(config.Project{})
+	ctx.SkipPublish = true
+	ctx.Artifacts.Add(&artifact.Artifact{
+		Name:   "mybin",
+		Path:   "nope.snap",
+		Goarch: "amd64",
+		Goos:   "linux",
+		Type:   artifact.PublishableSnapcraft,
+	})
+	testlib.AssertSkipped(t, Pipe{}.Publish(ctx))
 }
 
 func TestDefaultSet(t *testing.T) {

